docs/Activation-Recovery.md

```diff
@@ -0,0 +1,234 @@
+- **REVOKED** – the recovery code record is revoked for another reason (associated activation is removed, bank account is closed, etc...) and cannot be used for recovery activation.
+
+The **initial state** of the record depends on how's the source of the code:
+
+- Recovery Code created as a part of application activation is in **ACTIVE** state.
+- Recovery Code created for the [Recovery Postcard](Recovery-Postcard.md) is in **CREATED** state and needs to be [confirmed](#recovery-code-confirmation) before use. 
+
+The record also contains the following additional important information: 
+
+- **User Identifier** – links this record to one particular user in the system
+- **Number of failed attempts** – is a counter of failed attempts. The counter is increased when a wrong Recovery PUK is used and zeroed in case of correct one. If the counter reaches the maximum allowed numbers of failed attempts, then the record for given Recovery Code is **BLOCKED**
+- (optional) **Activation Identifier** – identifier for associated activation. The value is set in case that record is created as a part of application activation. 
+
+### Recovery PUK States
+
+Record associated with Recovery PUK has its own states:
+
+- **VALID** – the recovery PUK record is valid and ready to be used for a recovery operation.
+- **USED** – the recovery PUK record was already used for a recovery operation.
+- **INVALID** – the recovery PUK record is no longer valid because the related recovery code was either blocked or revoked.
+
+
+## Recovery Code Confirmation
+
+The Recovery Code printed on the [Recovery Postcard](Recovery-Postcard.md) has to be confirmed before it can be used for a recovery activation. The confirmation can be done in the PowerAuth Client application once is activated. The reason for that is that the issuer of the postcard has to be sure, that the postcard was delivered to the user.
+
+### Confirmation User Flow
+
+From the user's perspective, the confirmation process is quite simple:
+
+1. In the PowerAuth Client, the user selects the "Confirm Recovery Code" option.
+
+1. The user scans "Recovery QR Code" or enter the code manually. The PowerAuth Client application can detect whether the QR code contains valid Recovery Code (see [format](#format-of-recovery-code))
+
+1. The PowerAuth Client application sends an appropriate HTTP request to the Intermediate Server Application to confirm the code. 
+
+1. If the confirmation succeeded, the response from the server contains information, whether the Recovery Code was already confirmed before. The PowerAuth Client application can use this information to adjust the success screen presented to the user.
+
+Due to testing purposes, we decided that also Recover Codes created as a part of application activation can be confirmed. We're using this feature for the integration tests, between [PowerAuth mobile SDK](https://github.com/wultra/powerauth-mobile-sdk) and the Intermediate Server Application.
```

Comment:
Just a stylistic issue but maybe it would be better to change the structure.

"Recovery codes created during the activation can be also confirmed. However, this feature is mostly important for testing purposes, for integration tests between ..."